tools/cloud-build/daily-tests/ansible_playbooks/tasks/gather_logs.yml

```diff
@@ -0,0 +1,31 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+- name: Assert variables are defined
+  ansible.builtin.assert:
+    that:
+    - terraform_apply_stderr is defined
+
+- name: Remove New Lines From Terraform Apply STDERR
+  ansible.builtin.set_fact:
+    terraform_apply_stderr_one_line: "{{ terraform_apply_stderr | replace('\n',' ') }}"
+- name: Get Startup Script Logs
+  ansible.builtin.command: "{{ terraform_apply_stderr_one_line | regex_search('please run: (.+)', '\\1') | first }}"
```

Comment:
We are still prone to hitting the same error if for some reason wait for startup does not spit out the gcloud command. In the current ordering this error will happen after destruction but as pointed out above, this ordering means we will never get the logs.

When looking at your example builds it seems to me that the real issue is in the output generated by wait-for-startup. In the failure case wait-for-startup is generating:

```
could not detect end of startup script. Sleeping.
invalid return status '2'
to inspect the startup script output, please run:


MSG:
```

Why is it not printing the command after `please run:`?

Ideally we would answer that question and solve that problem as that is a user facing issue.

We we are not able to resolve that then instead I would suggest making this jinja expression more resilient. Ideally it would not fail if the regex does not find anything, although again this is surfacing a user facing error with wait-for-startup so I kind of want it to fail.

Reply:
This PR is deliberately not intended to address the specific issue in the gcloud command but to solve the "prone-to-leak" problem of the current test implementation. A follow-up PR is planned to address the stdout issue that just happened to trigger the problem.

Reply:
As a bread crumb. Root cause was addressed in: https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/1238